app/src/main/java/ch/epfl/sweng/rps/services/OfflineGameService.kt

```diff
@@ -0,0 +1,111 @@
+package ch.epfl.sweng.rps.services
+
+import ch.epfl.sweng.rps.models.ComputerPlayer
+import ch.epfl.sweng.rps.models.Game
+import ch.epfl.sweng.rps.models.Hand
+import ch.epfl.sweng.rps.models.Round
+import com.google.firebase.Timestamp
+import com.google.firebase.firestore.FirebaseFirestoreException
+import java.util.*
+
+class OfflineGameService(
+    private val uid: String,
+    private val computerPlayer: ComputerPlayer,
+    private var roundCount: Int
+) : GameService {
+
+    private var _game: Game? = null
+    private var _round: Round? = null
+    private var computerCurrentHand: Hand? = null
+    private var userCurrentHand: Hand? = null
+
+    /**
+     * Initialises the game after the game choice and play again button.
+     * (play again currently not available)
+     */
+    fun createGame() {
+        // gameMode is ignored
+        // done in not clear
+        // player count hardcoded, no support for different modes now
+        _game = Game(
+            uid,
+            listOf(computerPlayer.computerPlayerId),
+            mutableMapOf(),
+            1,
+            "",
+            false,
+            Timestamp(Date()),
+            2
+        )
+    }
+
+    //TODO: change when support with db is added
+    override val gameId: String
+        get() = _game!!.current_round.toString()
+
+    override val isGameOver: Boolean
+        get() = _game?.current_round == roundCount
+
+
+    suspend fun determineRoundResult(userHand: Hand): Hand.Result {
+        userCurrentHand = userHand
+        computerCurrentHand = computerPlayer.makeMove()
+        _round = addRound()
+        val scoreList = _round!!.computeScores()
+        var currentUserResult: Hand.Result? = null
+        for (scoreRes in scoreList) {
+            if (scoreRes.uid == uid) {
+                currentUserResult = scoreRes.results[0]
+                break
+            }
+        }
+        return currentUserResult!!
+    }
```

Comment:
This method shouldn't have side effects, only methods from GameService should drive the game. Notably 
```kotlin
userCurrentHand = userHand
computerCurrentHand = computerPlayer.makeMove()
_round = addRound()
```
 shouldn't be there.

Reply:
Okay, now I'm trying to change it though there are no methods in GameService that would let me do the computation and determine the winner. How do you want me to proceed?

Reply:
It's not up to the GameService to do so. All the data should be in the Game object, thus making the score computable however we need later on.